packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.tsx

```diff
@@ -0,0 +1,369 @@
+
+  /**
+   * Optional action button(s) to display at the bottom of the list.
+   */
+  readonly children?: ReactElement | ReactElement[];
+
+  /**
+   * Placeholder text to display in the search input. Defaults to "Search".
+   */
+  readonly searchPlaceholder?: string;
+
+  /**
+   * pre selected option
+   * @default ""
+   * @type string
+   */
+  readonly selected: ComboboxOption | null;
+
+  /**
+   * The encapsulating noun for the content of the combobox. Used
+   * in the empty state, and search placeholder. Should be pluralized.
+   */
+  readonly subjectNoun?: string;
+}
+
+export function ComboboxContent(props: ComboboxContentProps): JSX.Element {
+  const optionsExist = props.options.length > 0;
+  const {
+    searchValue,
+    setSearchValue,
+    selectedElement,
+    setSelectedElement,
+    open,
+    setOpen,
+    popperRef,
+    popperStyles,
+    attributes,
+    filteredOptions,
+    optionsListRef,
+  } = useComboboxContent(props.options);
+
+  useComboboxAccessibility(
+    open,
+    handleSelection,
+    popperRef,
+    filteredOptions,
+    optionsListRef,
+    selectedElement,
+  );
+
+  const template = (
+    <div
+      ref={popperRef}
+      id="ATL-Combobox-Content"
+      data-testid="ATL-Combobox-Content"
+      tabIndex={0}
+      className={classnames(styles.content, !open && styles.hidden)}
+      style={popperStyles.popper}
+      {...attributes.popper}
+    >
+      <Search
+        open={open}
+        placeholder={props.subjectNoun}
+        searchValue={searchValue}
+        setSearchValue={setSearchValue}
+      />
+      <ul className={styles.optionsList} role="listbox" ref={optionsListRef}>
+        {optionsExist &&
+          filteredOptions.map(option => {
+            const isSelected =
+              option.id.toString() === props.selected?.id.toString();
+
+            return (
+              <li
+                ref={listItem => {
+                  if (isSelected) setSelectedElement(listItem);
+                }}
+                key={option.id}
+                tabIndex={-1}
+                role="option"
+                aria-selected={isSelected}
+                onClick={() => handleSelection(option)}
+                className={classnames(styles.option, {
+                  [styles.selectedOption]: isSelected,
+                })}
+              >
+                {option.label}
+                {isSelected && <Icon name="checkmark" color="blue" />}
+              </li>
+            );
+          })}
+
+        {optionsExist && filteredOptions.length === 0 && (
+          <div className={styles.filterMessage}>
+            <Text variation="subdued">No results for {`“${searchValue}”`}</Text>
+          </div>
+        )}
+
+        {!optionsExist && (
+          <div className={styles.emptyStateMessage}>
+            <Text variation="subdued">
+              {getZeroIndexStateText(props.subjectNoun)}
+            </Text>
+          </div>
```

Comment:
not sure about having these two divs inside the `ul` as direct children. I do not believe that is a best practice

>[Content model](https://html.spec.whatwg.org/multipage/dom.html#concept-element-content-model):
Zero or more [li](https://html.spec.whatwg.org/multipage/grouping-content.html#the-li-element) and [script-supporting](https://html.spec.whatwg.org/multipage/dom.html#script-supporting-elements-2) elements.
https://html.spec.whatwg.org/multipage/grouping-content.html#the-ul-element

Reply:
Update here: We discussed adjusting this in Zak's current PR for `ComboboxContent` refactor